src/cpu/arm.rs

```diff
@@ -172,13 +180,46 @@ fn detect_features() -> u32 {
     features
 }
 
+#[cfg(all(target_os = "ios", target_arch = "aarch64"))]
+fn detect_features() -> u32 {
+    let mut features = ARMCAP_STATIC;
+
+    extern "C" {
+        fn sysctlbyname(
+            name: *const u8,
+            oldp: *mut core::ffi::c_void,
+            oldlenp: *mut crate::c::size_t,
+            newp: *mut core::ffi::c_void,
+            newlen: crate::c::size_t,
+        ) -> core::ffi::c_int;
+    }
+
+    let mut value: core::ffi::c_int = 0;
+    let mut len = core::mem::size_of_val(&value);
+    let rc = unsafe {
+        sysctlbyname(
+            "hw.optional.armv8_2_sha512\0".as_ptr(),
+            &mut value as *mut _ as _,
+            &mut len,
+            core::ptr::null_mut(),
+            0,
+        )
+    };
+    if rc == 0 && len == core::mem::size_of_val(&value) && value != 0 {
```

Comment:
It would be better for us to have wrapper for `sysctlbyname` that looks something like this:

```
fn detect_feature(null_terminated_name: &[c_char]) -> bool {
    debug_assert_eq!(null_terminated_name.last(), Some(0));
    let mut value = 0;
    let mut len = core::mem::size_of_val(&value);
    ... yada yada ...
    if rc != 0 {
      return false;
   }
   if len != size_of_val(&value) {
     return false; // unreachable
   }
   value != 0
}
```
This will allow code coverage measurement to show us in more detail if we're triggering the error/impossible condition(s).